src/YouTubePlayer.js

```diff
@@ -33,21 +34,80 @@ YouTubePlayer.proxyEvents = (emitter) => {
+ * @param {boolean} strictState A flag designating whether or not to wait for
+ * an acceptable state when calling supported functions. Default: `false`.
  * @returns {Object}
  */
-YouTubePlayer.promisifyPlayer = (playerAPIReady) => {
+YouTubePlayer.promisifyPlayer = (playerAPIReady, strictState = false) => {
   const functions = {};
 
   _.forEach(functionNames, (functionName) => {
-    functions[functionName] = async (...args) => {
-      const player = await playerAPIReady;
-
-      // TRICKY: Just spread the args into the function once Babel is fixed:
-      // https://github.com/babel/babel/issues/4270
-      //
-      // eslint-disable-next-line prefer-spread
-      return player[functionName].apply(player, args);
-    };
+    if (strictState && FunctionStateMap[functionName] instanceof Object) {
+      functions[functionName] = async (...args) => {
+        const stateInfo = FunctionStateMap[functionName];
+        const player = await playerAPIReady;
+        const playerState = player.getPlayerState();
+
+        // TRICKY: Just spread the args into the function once Babel is fixed:
+        // https://github.com/babel/babel/issues/4270
+        //
+        // eslint-disable-next-line prefer-spread
+        const value = player[functionName].apply(player, args);
+
+        // TRICKY: For functions like `seekTo`, a change in state must be
+        // triggered given that the resulting state could match the initial
+        // state.
+        if (
+          stateInfo.stateChangeRequired ||
+
+          // eslint-disable-next-line no-extra-parens
+          (
+            stateInfo.acceptableStates instanceof Array &&
+            stateInfo.acceptableStates.indexOf(playerState) === -1
+          )
+        ) {
+          await new Promise((resolve) => {
+            const onPlayerStateChange = () => {
+              const playerStateAfterChange = player.getPlayerState();
+
+              let timeout;
+
+              if (typeof stateInfo.timeout === 'number') {
+                timeout = setTimeout(() => {
+                  player.removeEventListener('onStateChange', onPlayerStateChange);
+
+                  resolve();
+                }, stateInfo.timeout);
+              }
+
+              if (
+                stateInfo.acceptableStates instanceof Array &&
+                stateInfo.acceptableStates.indexOf(playerStateAfterChange) !== -1
+              ) {
+                player.removeEventListener('onStateChange', onPlayerStateChange);
+
+                clearTimeout(timeout);
+                resolve();
+              }
+            };
+
+            player.addEventListener('onStateChange', onPlayerStateChange);
+          });
+        }
+
+        return value;
+      };
+    } else {
+      functions[functionName] = async (...args) => {
+        const player = await playerAPIReady;
+
+        // TRICKY: Just spread the args into the function once Babel is fixed:
```

Comment:
In this case, you're right, it should just be a TODO. But some times, when it is not obvious what code does or why, you use `TRICKY:` to annotate that fact while expounding upon what is really happening.